### PR TITLE
database: fix error handling in GetLabelOfLatestEE

### PR DESCRIPTION
**Pull request by jvehent**

### CODE REVIEW

File: database/connect.go [hunk 1]
```diff
@@ -69,7 +69,11 @@ func Connect(config Config) (*Handler, error) {
 	if config.MaxIdleConns > 0 {
 		dbfd.SetMaxIdleConns(config.MaxIdleConns)
 	}
-	return &Handler{dbfd}, nil
+	h := &Handler{dbfd}
+	dbCheckCtx, dbCancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (g-k) [inline database/connect.go:73]: +1 for adding a timeout

File: database/queries.go [hunk 2]
```diff
@@ -53,8 +53,11 @@ func (db *Handler) GetLabelOfLatestEE(signerID string, youngerThan time.Duration
 				WHERE is_current=TRUE AND signer_id=$1 AND created_at > $2
 				ORDER BY created_at DESC LIMIT 1`,
 		signerID, maxAge).Scan(&label, &nullableX5U)
-	if err == sql.ErrNoRows {
-		return "", "", ErrNoSuitableEEFound
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return "", "", ErrNoSuitableEEFound
+		}
+		return
```
Comment (g-k) [inline database/queries.go:60]: ah non-`ErrNoRows` errors weren't returned

File: docker-compose.yml [hunk 3]
```diff
@@ -49,6 +49,8 @@ services:
     volumes:
       - hsmtmpdir:/tmp/
       - hsmconfig:/app/
+    depends_on:
+      - db
```
Comment (g-k) [inline docker-compose.yml:53]: :+1: 

does this let us drop the wait-for-it scripts?

Comment (jvehent) [reply docker-compose.yml:53]: I don't think so. It'll just start the container after the DB, but provisioning might take time, so the `wait-for-it` script is still needed.

**Pull request merged**